source/common/crypto/crypto_impl.h

```diff
@@ -13,7 +13,17 @@ class PublicKeyObject : public Envoy::Common::Crypto::CryptoObject {
 public:
   PublicKeyObject() = default;
   PublicKeyObject(EVP_PKEY* pkey) : pkey_(pkey) {}
-  PublicKeyObject(const PublicKeyObject& pkey_wrapper);
+  EVP_PKEY* getEVP_PKEY() const;
+  void setEVP_PKEY(EVP_PKEY* pkey);
+
+private:
+  bssl::UniquePtr<EVP_PKEY> pkey_;
+};
+
+class PrivateKeyObject : public Envoy::Common::Crypto::CryptoObject {
```

Comment:
This is identical to `PublicKeyObject`. Maybe just rename `PublicKeyObject` to `PKeyObject` and not duplicate.